pkg/tsdb/influxdb/response_parser.go

```diff
@@ -19,7 +19,7 @@ var (
 	legendFormat = regexp.MustCompile(`\[\[([\@\/\w-]+)(\.[\@\/\w-]+)*\]\]*|\$(\s*([\@\w-]+?))*`)
 )
 
-func (rp *ResponseParser) Parse(buf io.ReadCloser, query *Query) *backend.QueryDataResponse {
+func (rp *ResponseParser) Parse(buf io.ReadCloser, queries []Query) *backend.QueryDataResponse {
 	resp := backend.NewQueryDataResponse()
 	queryRes := backend.DataResponse{}
```

Comment:
i recommend moving this line `queryRes := backend.DataResponse{}` inside the for-loop,
otherwise we are reusing this object and just adjusting it's fields every time, and things could get mixed up.